### PR TITLE
net: lwm2m: Dynamic resource instance allocate support and integrate

### PR DESCRIPTION
**Pull request by jheiskan81**
- Added support for allocate dynamically resource instance.
- Integrated Dynamic resource allocation to: JSON, Plain text & OMA TLV

### CODE REVIEW

**Comment by jheiskan81** [comment]
@SeppoTakalo , @VeijoPesonen , @jarlamsa  and @rlubos Can you review this

**Review by SeppoTakalo — APPROVED** [review]

**Review by VeijoPesonen — APPROVED** [review]

File: subsys/net/lib/lwm2m/lwm2m_engine.c [hunk 1]
```diff
@@ -2018,9 +2018,69 @@ void lwm2m_engine_get_binding(char *binding)
...
+	}
+
+	if (!r) {
+		return -ENOENT;
+	}
+	/* Store resource pointer */
+	*res = r;
+
+	if (!r_i) {
+		if (path->level < 4) {
```
Comment (rlubos) [inline subsys/net/lib/lwm2m/lwm2m_engine.c:2066]: `path->level < LWM2M_PATH_LEVEL_RESOURCE_INST`

File: subsys/net/lib/lwm2m/lwm2m_rw_json.c [hunk 2]
```diff
@@ -910,7 +910,7 @@ int do_write_op_json(struct lwm2m_message *msg)
 	struct lwm2m_engine_res_inst *res_inst = NULL;
```
Comment (rlubos) [inline subsys/net/lib/lwm2m/lwm2m_rw_json.c:910]: The JSON commit has duplicated `net: lwm2m:` prefix in the title.

**Review by rlubos — COMMENTED** [review]
Nice, I like that the patch removes more lines than it adds, still introducing new functionality. Just some nits.

File: subsys/net/lib/lwm2m/lwm2m_engine.c (hunk 1 above)
Comment (jheiskan81) [reply subsys/net/lib/lwm2m/lwm2m_engine.c:2066]: I will change

File: subsys/net/lib/lwm2m/lwm2m_rw_json.c (hunk 2 above)
Comment (jheiskan81) [reply subsys/net/lib/lwm2m/lwm2m_rw_json.c:910]: I will fix that commit message.

**Comment by jheiskan81** [comment]
@rlubos Thank you for good spot. I have fix your comment's.

**Review by rlubos — APPROVED** [review]

**Pull request merged**